### PR TITLE
DOC: GSoC Blogs Week 6, 7, 8

### PR DESCRIPTION
**Pull request by robinroy03**
GSoC Blogs Week 6, 7, 8

Preview:
| Week 6 | Week 7 | Week 8 |
|---|---|---|
| image | image | image |

### CODE REVIEW

File: docs/source/posts/2024/2024-07-27-week6-robin.rst [hunk 1]
```diff
@@ -0,0 +1,49 @@
+Week 6: UI Improvements and RAG performance evaluation
```
Comment (WassCodeur) [inline docs/source/posts/2024/2024-07-27-week6-robin.rst:1]: Hi @robinroy03! 

Great jobs

File: docs/source/posts/2024/2024-07-27-week7-robin.rst [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+====================================
+
+.. post:: July 27 2024
+   :author: Robin Roy
+   :tags: google
+   :category: gsoc
+
+Hi, I'm `Robin <https://github.com/robinroy03>`_ and this is my blog about week 7.
+
+I majorly took this week off due to my semester final examinations :) They were fun. Major topics were x86, ARM and 8051. I had not written a lot of assembly apart from school work. I took the week to experiment with some assembly. The course was more into hardware architecture than programming. I've now enough knowledge to read a given piece of ASM code with a wiki to look up mnemonics (and Gemini/Claude to help). I'm not fast in writing ASM (yet), one day I'll find a project to dive into, or maybe some reverse engineering and CTFs. GPU instruction sets are also something interesting.
```
Comment (WassCodeur) [inline docs/source/posts/2024/2024-07-27-week7-robin.rst:11]: Congratulations on completing your semester exams! 🎉

File: docs/source/posts/2024/2024-07-27-week8-robin.rst [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+---------------------------
+
+No, I did not get stuck anywhere.
+
+LINKS:
+
+- `Parameter-Efficient Fine-Tuning <https://huggingface.co/blog/peft>`_
+- `catastrophic forgetting <https://arxiv.org/abs/1312.6211>`_
+
+Thank you for reading!
```
Comment (WassCodeur) [inline docs/source/posts/2024/2024-07-27-week8-robin.rst:52]: LGTM, Thank you for sharing.

**Review by WassCodeur — APPROVED** [review]

**Review by skoudoro — APPROVED** [review]
ok, blogs looks good, I am go ahead and merge

**Pull request merged**